lib/chef/provider/execute.rb

```diff
@@ -58,7 +58,16 @@ def action_run
         end
 
         converge_by("execute #{description}") do
-          result = shell_out!(command, opts)
+          begin
+            shell_out!(command, opts)
+          rescue Mixlib::ShellOut::ShellCommandFailed
+            if sensitive?
+              raise Mixlib::ShellOut::ShellCommandFailed,
```

Comment:
This also modifies the stack trace and such, but given that we're only 2 lines away from where the error must have been raised in the first place, it's probably not a big deal.